### PR TITLE
Kamal Deployment (WIP)

### PR DESCRIPTION
**Pull request by edwinthinks**
This is the start of deploying to Azure using Kamal 2.0. This is still WIP and I need to update the README plus add details on why some things were changed.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -57,7 +60,6 @@ gem "devise"
 gem "devise_invitable", "~> 2.0.9"
 
 # Use Sass to process CSS
-gem "dartsass-sprockets"
```
Comment (edwinthinks) [inline Gemfile:60]: Removed to have propshaft make the deployment easier

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -4,8 +4,7 @@
 @import "rails_bootstrap_forms";
 
 // Geek theme (https://geeksui.codescandy.com/)
-@import 'theme.min.css';
-@import 'feather.css';
+@import 'theme.min.scss';
```
Comment (edwinthinks) [inline app/assets/stylesheets/application.scss:7]: Dartsass compiles this `theme.min.scss` and puts it in app/assets/builds which we allow access via propshaft

File: app/controllers/root_controller.rb [hunk 3]
```diff
@@ -1,10 +1,15 @@
 class RootController < ApplicationController
   skip_before_action :authenticate_user!
-  skip_verify_authorized only: %i[index]
+  skip_verify_authorized only: %i[index up]
```
Comment (edwinthinks) [inline app/controllers/root_controller.rb:3]: Health check that is necessary to deploy! A simple `plain: ok` is good enough. Make sure to have no authorization requirements.

File: bin/docker-entrypoint [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash -e
```
Comment (edwinthinks) [inline bin/docker-entrypoint:1]: Added because it is needed for deploying

File: config/application.rb [hunk 5]
```diff
@@ -28,11 +28,10 @@ class Application < Rails::Application
     #
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
+    config.assets.paths << Rails.root.join("app", "assets", "builds")
```
Comment (edwinthinks) [inline config/application.rb:31]: Propshaft related to read the build output of dartsass

File: config/credentials/production.yml.enc [hunk 6]
```diff
@@ -0,0 +1 @@
+N3M0tKvmR0uUcdEp/oNA02rHCOXwZY2CC/lvtbie1btkT8bjzx6o3gYTce/XjUqZbqTydgi/TtwZFspNaMA2wMTlxBwEGLwFY+WqFwBKHFK1cw0HqRqAVspVOTRHHYURnzzxt3IqpaQBm/pR4g6ypqa4q0vQlYFBow6EZOvI0XQckdcoqIZvrd3l5GQuKXBpTAHfIcES/NDCUN8rPuhEyOtzHvgIMTRkoEPK21nVtM8MC8ZM8aPh/xpysIUdyJ9M09Hyd2Q+B96RMdmPV5IQvZUwePjun9IxOb4lyIen6iI49hXjimo6W0ct6cXKhLFF9n9YyInYwZcZlLkE3OThiZsgqUluM/hxz93Cn9De9pF85/Ju/UpfqNdYuRbOo1IRMizSYbXihb3UhUwaSIo2TWhoH9y6zCTFqU7fPOK+K4tr3hPGyQxSQaqol0+yiFOlWNGs3UtFhZtUILkqV1WBaDelwzW92gQGEqK7RteXaFVrAVFbbftwdgT1jJgSvLTHml6ysnOZNRwjeskiZWxHa2DhiKQQb81b//r5ma7LPU7Nfli0IyMokbWumzN1cTq44bjTk5qSa0lzREtLGkKFLVJiP+HJ/EaHu+H+DRBlRVQYO1G4vZ4ECBJzcfFxpU1ClyPbwf4IurW/--ZmNjs3R4MOx9x1hJ--ArdhnAr6jbIXpqgKEIcsyQ==
```
Comment (edwinthinks) [inline config/credentials/production.yml.enc:1]: New goodness for the credentials for deployment. Please update this when we go real production.

**Comment by kasugaijin** [comment]
@edwinthinks @jmilljr24 I think we can close this PR now that Justin has built off of it and made this https://github.com/rubyforgood/homeward-tails/pull/1117 ?

**Comment by kasugaijin** [comment]
I will close this now as we have merged in deployment in a separate PR. Thanks again @edwinthinks !!!

**Pull request closed without merging**